src/net_packet.c

```diff
@@ -67,23 +67,10 @@ static void send_mtu_probe_handler(event_loop_t *loop, void *data) {
 		return;
 	}
 
-	if(n->mtuprobes > 32) {
-		if(!n->minmtu) {
-			n->mtuprobes = 31;
-			timeout = mesh->pinginterval;
-			goto end;
-		}
-
-		logger(mesh, MESHLINK_INFO, "%s did not respond to UDP ping, restarting PMTU discovery", n->name);
-		n->status.udp_confirmed = false;
-		n->mtuprobes = 1;
-		n->minmtu = 0;
-		n->maxmtu = MTU;
-	}
-
-	if(n->mtuprobes >= 10 && n->mtuprobes < 32 && !n->minmtu) {
+	if(n->mtuprobes >= 10 && n->mtuprobes < 30 && !n->minmtu) {
 		logger(mesh, MESHLINK_INFO, "No response to MTU probes from %s", n->name);
-		n->mtuprobes = 31;
+		n->mtuprobes = 10;
```

Comment:
But this means that if we don't get a reaction to PMTU probes, we keep sending a burst of three random sized packets, but now just every `mesh->pinginterval`. I think in this case we should also send the smallest possible probe; when we get a response we will restart the PMTU discovery from scratch anyway.